iOS/Source/SunView.swift

```diff
@@ -147,15 +125,23 @@ class SunView: UIView {
         return SunViewLocation(x: absoluteX, y: absoluteY)
     }
 
-    func updateInterface(withBackgroundColor backgroundColor: UIColor, textColor: UIColor, andPercentageInDay percentageInDay: Double, sunPhase: SunTime.SunPhase) {
-        self.sunriseLabel.textColor = textColor
-        self.sunsetLabel.textColor = textColor
-        self.currentTimeLabel.textColor = textColor
-        self.horizon.backgroundColor = textColor
-        self.sun.tintColor = textColor
-        self.moon.backgroundColor = backgroundColor
+    func updateInterface(controller: DaylightModelController) {
+        self.sunriseLabel.textColor = controller.secondaryColor
+        self.sunsetLabel.textColor = controller.secondaryColor
+        self.currentTimeLabel.textColor = controller.secondaryColor
+        self.horizon.backgroundColor = controller.secondaryColor
+        self.sun.tintColor = controller.secondaryColor
+        self.moon.backgroundColor = controller.primaryColor
+
+        self.sunViewLocation = self.location(for: controller.percentageInDay)
+        if controller.shouldShowMoon {
+            self.moon.isHidden = false
+            self.sunViewLocation = SunViewLocation(x: (self.frame.width - SunView.sunSize) / 2.0, y: 0.0)
```

Comment:
Should this be adjusting the moon view's location instead of the sun view's?

Reply:
No, i needed to wrap my head around it as well, but apparantly the moon is just a view that blocks a part of the sun. but i mean to rewrite this!